### DISCUSSION TITLE
Return on Investment

### DISCUSSION THREAD

**Discussion by jheredianet** [jheredianet/Teslamate-CustomGrafanaDashboards#81] [Ideas]
Hi, @stauffenberg2020 

I'd like to ask for your help about your dashboard. Please, could you answer the following questions?

1. About Depreciated Value Factor = 0.786 - 0.002363 *(Years * 12) Where do the fixed values come from? Are there related to a county or anything similar? or is there a pattern for that? Is it OK if we leave the user to modify it dynamically?
2. Could you describe the meaning of these columns? cum_el_cost, period_el_cost, car_value_pls_el_savings

I'll let you know the changes I'm developing when ready. Thanks in advance for your help.

**Comment by stauffenberg2020** [comment]
Hi @jheredianet,

1. I wanted to keep the Depreciation value factor as a user input, But I didnt know how to do that. I am also thinking that the depreciated value will be country/market specific and will depend on the age and mileage of the car. Since I do not country/market specifications, etc - I simply assumed a generalised equation/function for calculating the depreciated value after months/years.
2. Let me explain each of the columns

- "period" can be selected as either 'month' or 'year' in the drop down at the top of the dashboard. So if selected as 'month', its going to be the data accumulated from 1st month at which TeslaMate started logging data, So 1,2,3 will be the month numbers. If "period" is selcted as 'year', then its the year numbers
- "periodic_mileage" is the number of kms/miles the car has driven in that period. Lets say March 2024, Its the number of Kms/miles driven in that month
- "period_el_cost" is the cost spent for electricity in that particular period. Lets say its the total money spent in entire March 2024
- "period_cost_per_km"  is "period_el_cost"/"periodic_mileage"
- "cum_mileage" is the cumulative mileage upto that perticular month. For eg, if for February 2024, if the car has driven 500 Kms and for March 2024, if the car has driven 800 Kms, then the "cum_mileage" for February 2024 will be 500 and for March 2024 will be 500+800 = 1300 Kms
- "cum_el_cost" is the cumulative cost similar to "cum_mileage"
- "cum_el_savings" is running cost savings for driving an electric car when compared to a fossil fuel car. Unfortunately I hardcoded it as a country (Denmark) specific one. Let me explain in detail. In Denmark, If someone is driving a petrol car, he will roughly spend 1 DKK per Km. So if someone is driving 500 Kms a month, He will have to roughly spend 500 DKK for fuel.... So the "cum_el_savings" (for Denmark) is "cum_mileage"*1(DKK/Km) - "cum_el_cost". But I guess this should be considered as a user input
- "depreciated_car_value" is the current cost of the car when the person is selling it back
- "car_value_pls_el_savings" is simply adding "depreciated_car_value" + "cum_el_savings". So the expectation is like when one can get the Break-Even for the money we put into buying the car by driving at a cheaper electricity cost when compared to driving a fossil fuel car 

I hope it is somewhat clear now. I am sorry if I have made it complicated. But please reply me whenever you need more clarification.

**Reply by jheredianet** [reply]
Great! Thank you very much. It clarifies everything. I will make some changes and let you know to test if it works as expected. Let's be in touch.

**Comment by jheredianet** [comment]
@stauffenberg2020 Just one more question about Depreciated Value Factor = 0.786 - 0.002363 *(Years * 12)

What do 0.786 and 0.002363 mean? I guess they are two different variables, and like to know their names or what do they represent.

**Reply by stauffenberg2020** [reply]
Hi @jheredianet, I have assumed "what would be the depreciated value after driving the car 20000 Kms each year" and then fitted an equation for my assumed values. Thats how I got the above equation. But again - this is my simple assumption and it must be a user input, Because the depreciation % will depend on a lot of parameters like current condition of the car, country/market demands, mileage of the car, etc. 

Please find an excel in the attachment where I show how I got the above equation. Those two numbers are slope (-0.002363) and intercept (0.786) of a linear fit (for the assumed values in the excel). 

Like y = mx+c, where m is the slope and c is the intercept.

[tesla_depreciation.xlsx](https://github.com/jheredianet/Teslamate-CustomGrafanaDashboards/files/15052208/tesla_depreciation.xlsx)

Please let me know if you need any other assistance from my side

**Reply by jheredianet** [reply]
Hi @stauffenberg2020 
First of all sorry, I was very busy to attend this. 

I've have made a preview version that I'd like you to take a look and let me know what you think. 

It's still in a special branch 
https://github.com/jheredianet/Teslamate-CustomGrafanaDashboards/blob/feature/Return_on_Investment/dashboards/ReturnOnInvestment.json

When it's validated by you, I'll merge it to the main branch. 

Please feel free to make a pull request on that specific branch if you consider to make some changes. 
Since this estimation needs many understanding about the estimations I've made a Help section below the table. So, any use could understand how to use it. 

Let me know what you think. 

Regards.

**Reply by stauffenberg2020** [reply]
Hi @jheredianet, I just had the time to take a look at the dashboard.

I am fine with the overall change you have given especially to the "depreciation per age" and adding "depreciation per mileage" model along with the associated help descriptions. 

I am not able to exactly match the numbers due to the change in the "depreciation" model - But that should be fine I guess - Because all other numbers are matching well.

However, there are two small things and it would be good if we can handle it.

1. Can we somehow allow the user to type in the 0.20 per mile or km (& let the user handle the length conversion of the depreciation) as a user input in the top just next to the depreciation percentage by age?
2. When I change the period from "years" to "months" and vice versa - I was hoping that the new depreciation model chart/plot also to get updated. But it always shows with "months" in the x-axis. Is it something you can fix it?

What do you think?

**Reply by jheredianet** [reply]
Hi @stauffenberg2020 

I've made the changes....

> 1. Can we somehow allow the user to type in the 0.20 per mile or km (& let the user handle the length conversion of the depreciation) as a user input in the top just next to the depreciation percentage by age?

I agree with this, is better to let the use to setup that value.

> 2. When I change the period from "years" to "months" and vice versa - I was hoping that the new depreciation model chart/plot also to get updated. But it always shows with "months" in the x-axis. Is it something you can fix it?

About this, I did it at the beginning but when the car es relatively new and you select by "year" as period, you'll only see 1 or 2 points only, so it's not representative. Anyway, I have made the change so that it is the user who decides how to see it.

Could you check it again?

**Reply by stauffenberg2020** [reply]
Hi @jheredianet, I have gone through the changes and it pretty fine. Please feel free to merge it to your main branch

**Reply by jheredianet** [reply]
Great!, I will change the title to "Amortization Tracker" because buying a car isn't really an investment  ;-)

**Comment by jheredianet** [comment]
Merged on the "Main" branch.
